MELOSYS-4254 - Lønn og godtgjørelser menypunkt

sigbjorn-myhre opened the pull request:
Menypunkt for lønn og godtgjørelser.

Skisser: https://www.figma.com/file/7valEmRFMgUIh4MIh7N0SC/Sidemeny?node-id=185%3A2797

Schema: https://github.com/navikt/melosys-schema/pull/216

sigbjorn-myhre committed:
Panel for lønn og godtgjørelser

sigbjorn-myhre committed:
Lint lønn og godtgjørelser

sigbjorn-myhre committed:
Lønn og godtgjørelser typer for komponenter

sigbjorn-myhre committed:
Lønn og godtgjørelser linting og tester

sigbjorn-myhre commented on src/ducks/behandlingsgrunnlag/operations.js:
```diff
@@ -94,6 +94,7 @@ const lagBehandlingsgrunnlagFelter = (behandlingsgrunnlag) => ({
 const lagSoeknadFelter = (behandlingsgrunnlag) => ({
   ...lagBehandlingsgrunnlagFelter(behandlingsgrunnlag),
   arbeidsinntekt: behandlingsgrunnlag.arbeidsinntekt,
+  loennOgGodtgjoerelse: behandlingsgrunnlag.loennOgGodtgjoerelse,
```
Har valgt å bruke `oe` for `ø` internt i frontend her siden det blir hentet fra backend. Vet at vi noen ganger "oversetter" til kun `o`. Har noen tanker om dette? Gjelder også feltene under lønn og godtgjørelse.

JohanSakshaug replied:
Tenker det er ok

sigbjorn-myhre commented on src/ducks/behandlingsgrunnlag/reducers.js:
```diff
@@ -242,6 +242,19 @@ export default function reducer(state = initialState, action) {
             })),
           },
+
+          loennOgGodtgjoerelse: {
```
Har ikke fjernet arbeidsinntekt fra frontend eller skjema, men lurer på om det også må gjøres, da den blir "erstattet" av denne.

JohanSakshaug replied:
Kan nok være lurt å fjerne hvis det ikke brukes ja 👍

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.tsx:
```diff
@@ -0,0 +1,342 @@
+
+const symbolsynlighetMap = new Map([
+  [Status.Redigerer, { bin: false, pencil: false }],
+  [Status.IngenData, { bin: false, pencil: false }],
```
Trenger ikke sette disse to, de blir satt slik som default. Tenkte at man skulle slippe å sette alle ved hver bruk av komponenten

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.tsx:
```diff
@@ -0,0 +1,342 @@
+  [Status.Redigerer, { bin: false, pencil: false }],
+  [Status.IngenData, { bin: false, pencil: false }],
+  [Status.RedigeringUtfort, { bin: false, pencil: true }],
+] as const);
```
Hva gjør `as const` her? Er det for å sette `readonly` på `bin` og `pencil`?

sigbjorn-myhre replied:
Ja. Det var egentlig bare noe jeg prøvde og glemte å fjerne igjen 😅

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.tsx:
```diff
@@ -0,0 +1,342 @@
+
+export const BooleanFeltRedigeringUtfort = ({ tekst, verdi }: BooleanFeltRedigeringUtfortProps) => (
+  <Nav.Row>
+    <Nav.Column xs="12" className="boolean-felt-redigerering-utfort">
```
redigerering -> redigering

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.less:
```diff
@@ -0,0 +1,49 @@
+  .boolean-felt-redigerering-utfort {
+    margin-bottom: 1em;
+
+    .typo-normal {
```
Jeg liker ikke så godt å være avhengig av interne css-klasser i nav-frontend. Tenker det er mer robust om `className`-prop på Nav.typo komponenten settes eksplisitt her(`className="typo-normal"` er ok).
Gjelder for typo-element også

sigbjorn-myhre replied:
Den er grei, skal ta en runde på det 👍

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.test.tsx:
```diff
@@ -0,0 +1,65 @@
+    });
+
+    it('viser "-" dersom verdi er tom streng', () => {
+      // @ts-ignore
```
Burde ikke typen til `props.verdi` tillate `string`, så man slipper `@ts-ignore`?

sigbjorn-myhre replied:
Jo, kan legge inn det

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.tsx:
```diff
@@ -0,0 +1,342 @@
+
+import "./lonnOgGodtgjorelser.css";
+
+type Redigerbart = { redigerbart: boolean };
```
Synes ikke det er nødvendig med en egen type for redigerbart her, og at det er mer lesbart uten. 👍 / 👎 ?

sigbjorn-myhre replied:
Var bare jeg som ble litt lei av å skrive `{ redigerbart: boolean };`
Jeg følte det ble mer lesbart i tillfeller som f.eks.
```ts
type ArbeidsgiveravgiftOgTrygdeavgiftProps = ArbeidsgiveravgiftOgTrygdeavgift & Redigerbart;
```
men for
```ts
type InntektRedigererProps = {
  tittel: string;
  feltNavn: string;
} & Redigerbart;
```
ble det kanskje mindre lesbart igjen 🤔

sigbjorn-myhre replied:
Endte opp med å fjerne den 👍

JohanSakshaug commented on src/felleskomponenter/menypanel/menypunkter/lonnOgGodtgjorelser/lonnOgGodtgjorelser.tsx:
```diff
@@ -0,0 +1,342 @@
+    harData
+    tittel={KV.Menypunkter.LonnOgGodtgjorelser.undertitler.lonnOgNaturalytelser}
+    hentNyStatusVedHarData={false}
+    onBinClick={() => {}}
```
Siden `bin`-ikon ikke vises i dette tilfellet, tenker jeg du kan sette `onBinClick` til optional i `EditerbartElement`

sigbjorn-myhre committed:
QA MELOSYS-4254

sigbjorn-myhre committed:
Fjerner arbeidsinntekt fra behandlingsgrunnlag

sigbjorn-myhre committed:
Fjerner Redigerbart-type

JohanSakshaug reviewed (approved):
👍  Gjenstår bare den bugen da..

sigbjorn-myhre committed:
Støtte for nullable lønn og godtgjørelser

- Fjerner to selectors som ikke ble brukt
- Oppdaterer typer i lonnOgGodtgjorelser.tsx da de ikke reflekterte at de er nullable

sigbjorn-myhre committed:
Oppdater behandlingsgrunnlag ved lagring av panel

sigbjorn-myhre committed:
Merge branch 'master' into MELOSYS-4254_lonn_og_godgjorelser
